notebooks/106-auto-device/106-auto-device.ipynb

```diff
@@ -0,0 +1,407 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "73d7aedb",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import cv2\n",
+    "import matplotlib.pyplot as plt\n",
+    "import numpy as np\n",
+    "from openvino.runtime import Core, CompiledModel\n",
+    "\n",
+    "ie = Core()\n",
+    "\n",
+    "model = ie.read_model(model=\"model/googlenet-v1.xml\")\n",
+    "compiled_model = ie.compile_model(model=model, config={\"LOG_LEVEL\":\"LOG_INFO\"})\n",
+    "\n",
+    "if isinstance(compiled_model, CompiledModel):\n",
+    "    print(\"Compile model without device_name successfully.\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "85b48949",
+   "metadata": {},
+   "source": [
+    "### compile_model with AUTO as device name"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0814073d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "compiled_model = ie.compile_model(model=model, device_name=\"AUTO\")\n",
+    "\n",
+    "if isinstance(compiled_model, CompiledModel):\n",
+    "    print(\"Compile model with AUTO successfully.\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "705ce668",
+   "metadata": {},
+   "source": [
+    "## First inference latency benifit with AUTO\n",
+    "One of the key performance benefits of AUTO is on first inference latency (FIL = compile model time + fist inference execution time). Directly using CPU device would produce the shortest first inference latency as the OpenVINO graph representations can really quickly be JIT-compiled to CPU. The challenge is with the GPU. Since the OpenCL complication of graph to GPU-optimized kernels takes a few seconds to complete for this platform. If AUTO selects GPU as the device, this initialization time may be intolerable to some applications, which is the reason for AUTO to transparently use the CPU as the first inference device until GPU is ready. \n",
+    "### Load an Image"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "dc1cfeaf",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "input_layer_ir = next(iter(compiled_model.inputs))\n",
+    "\n",
+    "# Text detection models expects image in BGR format\n",
+    "image = cv2.imread(\"data/intel_rnb.jpg\")\n",
+    "\n",
+    "# N, C,H,W = batch size, number of channels, height, width\n",
+    "N, C, H, W = input_layer_ir.shape\n",
+    "\n",
+    "# Resize image to meet network expected input sizes\n",
+    "resized_image = cv2.resize(image, (W, H))\n",
+    "\n",
+    "# Reshape to network input shape\n",
+    "input_image = np.expand_dims(resized_image.transpose(2, 0, 1), 0)\n",
+    "\n",
+    "plt.imshow(cv2.cvtColor(image, cv2.COLOR_BGR2RGB));"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f9fcaba9",
+   "metadata": {},
+   "source": [
+    "### compile_model with GPU Device and do first inference"
```

Comment:
Please remove "compile_model with" from any heading, leave it simple like "GPU device and first inference)